scbctl/cmd/scans.go

```diff
@@ -116,6 +126,74 @@ func NewScanCommand() *cobra.Command {
 	}
 
 	scanCmd.Flags().String("name", "", "Name of the created scan. If no name is provided, the ScanType will be used as the name")
+	scanCmd.Flags().Bool("follow", false, "Follow the log output")
 
 	return scanCmd
 }
+
+func followScanLogs(ctx context.Context, kubeclient client.Client, namespace, scanName string) error {
+	// Find the job associated with the scan
+	jobList := &batchv1.JobList{}
+	labelSelector := client.MatchingLabels{}
+
+	fmt.Println("Listing jobs in namespace:", namespace)
+
+	for {
+		fmt.Println("Attempting to list jobs...")
+		err := kubeclient.List(ctx, jobList, client.InNamespace(namespace), labelSelector)
+		if err != nil {
+			return fmt.Errorf("error listing jobs: %s", err)
+		}
+
+		if len(jobList.Items) == 0 {
+			fmt.Println("No jobs found, retrying...")
+			time.Sleep(2 * time.Second)
+			continue
+		}
+
+		fmt.Printf("Found %d job(s)\n", len(jobList.Items))
+
+		for _, j := range jobList.Items {
+			fmt.Printf("Job: %s, Labels: %v\n", j.Name, j.Labels)
+		}
+
+		var job *batchv1.Job
+		for _, j := range jobList.Items {
+			fmt.Printf(j.Name)
+			if strings.HasPrefix(j.Name, fmt.Sprintf("scan-%s", scanName)) {
```

Comment:
The better way to do this is (simmilarly to the cascading command) look for the job by using the owner references of the job to see which actually belong to the scan instead of doing a prefix matching.

Reply:
The hook status is for the hooks not the scan jobs.
I've changed the code to use the labels then the owner references if you want to have a look what i meant :)